solidity/src/proof_plans/SortMergeJoinExec.presl

```diff
@@ -0,0 +1,418 @@
+                    plan_ptr := add(plan_ptr, UINT64_SIZE)
+
+                    // We set to 0 any indices that are covered by the join columns
+                    mstore(add(other_indices, mul(join_column_index, WORD_SIZE)), 0)
+
+                    // Get the evaluation of the column from the input evaluations
+                    let join_column_eval := mload(add(evaluations, mul(join_column_index, WORD_SIZE)))
+
+                    // Both hat and join evaluations should store the join column evaluation
+                    mstore(target_hat_evals, join_column_eval)
+                    target_hat_evals := add(target_hat_evals, WORD_SIZE)
+                    mstore(target_join_evals, join_column_eval)
+                    target_join_evals := add(target_join_evals, WORD_SIZE)
+                }
+
+                // We need to iterate through each index to determine which were not covered by join columns
+                for { let i := 0 } lt(i, num_columns) { i := add(i, 1) } {
+                    // We only need to worry about the indices that were not join indices
+                    if mload(add(other_indices, mul(i, WORD_SIZE))) {
+                        // We store each column evaluation in the hat collection
+                        mstore(target_hat_evals, mload(add(evaluations, mul(i, WORD_SIZE))))
+                        target_hat_evals := add(target_hat_evals, WORD_SIZE)
+                    }
+                }
+                // The last entry in the hat collection is the rho evaluation
+                mstore(target_hat_evals, builder_consume_rho_evaluation(builder_ptr))
+                plan_ptr_out := plan_ptr
+            }
+            function evaluate_u_column_with_monotony_check(builder_ptr, alpha_and_beta) -> u_column_eval_array, u_chi_eval {
+                // The length of the u eval collection is 1, until we support joining on multiple columns
+                u_column_eval_array := mload(FREE_PTR)
+                mstore(FREE_PTR, add(u_column_eval_array, WORDX2_SIZE))
+                mstore(u_column_eval_array, 1)
+
+                // We run our monotony check on u eval, before wrapping it in the collection
+                u_chi_eval := builder_consume_chi_evaluation(builder_ptr)
+                let u_column_eval := builder_consume_first_round_mle(builder_ptr)
+                monotonic_verify(
+                    builder_ptr,
+                    mload(alpha_and_beta),
+                    mload(add(alpha_and_beta, WORD_SIZE)),
+                    u_column_eval,
+                    u_chi_eval,
+                    1,
+                    1
+                )
+
+                // Store the u eval in the collection
+                mstore(add(u_column_eval_array, WORD_SIZE), u_column_eval)
+            }
+            function consume_and_membership_check_left_column_evals(builder_ptr, alpha_and_beta, hat_evals, res_chi_eval, chi_eval) -> res_column_evals {
```

Comment:
Consume `res_u_column_evals`, `res_left_column_evals` and `rho_bar_left_eval` (from Rust) from left input with `left_rho` appended as last column. Then conduct membership check that enhanced left column parts of the join results are in fact rows of the enhanced left input.